generators/app/index.js

```diff
@@ -310,72 +309,186 @@ module.exports = JhipsterAuditGenerator.extend({
-            to: `${this.javaDir}web/rest/JaversEntityAuditResource.java`
-          });
-        }
-        if (this.enableTranslation) {
-          this.languages.forEach((language) => {
-            let sourceLanguage = 'en';
-            if (fs.existsSync(`${this.templatePath()}/src/main/webapp/i18n/${language}/entity-audit.json`)) {
-              sourceLanguage = language;
-            }
-            files.push({
-              from: `${this.webappDir}i18n/${sourceLanguage}/entity-audit.json`,
-              to: `${this.webappDir}i18n/${language}/entity-audit.json`
-            });
+            from: `${this.webappDir}i18n/${sourceLanguage}/entity-audit.json`,
+            to: `${this.webappDir}i18n/${language}/entity-audit.json`
           });
-        }
-        genUtils.copyFiles(this, files);
-        // add bower dependency required
+        });
+      }
+
+      genUtils.copyFiles(this, files);
+
+      // add bower dependency required
+      if (this.clientFramework === 'angular1') {
         this.addBowerDependency('angular-object-diff', '1.0.3');
         this.addAngularJsModule('ds.objectDiff');
-        // add new menu entry
-        this.addElementToAdminMenu('entity-audit', 'list-alt', this.enableTranslation, this.clientFramework);
-        if (this.enableTranslation) {
-          this.languages.forEach((language) => {
-            let menuText = 'Entity Audit';
-            try {
-              menuText = JSON.parse(fs.readFileSync(`${this.templatePath()}/src/main/webapp/i18n/${language}/global.json`, 'utf8')).global.menu.admin['entity-audit'];
-            } catch (e) {
-              this.log('Cannot parse file');
-            }
-            this.addAdminElementTranslationKey('entity-audit', menuText, language);
-          });
-        }
+      } else {
+        // add dependency required for displaying diffs
+        this.addNpmDependency('ng-diff-match-patch', '^2.0.6');
+        // based on BaseGenerator.addAdminToModule
+        const adminModulePath = `${this.webappDir}app/admin/admin.module.ts`;
+        this.rewriteFile(
+          adminModulePath,
+          'jhipster-needle-add-admin-module-import',
+          'import { DiffMatchPatchModule } from \'ng-diff-match-patch\';'
+        );
+        this.rewriteFile(
+          adminModulePath,
+          'jhipster-needle-add-admin-module',
+          'DiffMatchPatchModule,'
+        );
+
+        // register new component and service in admin module
+        // remove any previous references so we don't duplicate them
+        this.replaceContent(adminModulePath, /\s*EntityAuditComponent,/, '', true);
+        this.replaceContent(adminModulePath, /\s*EntityAuditModalComponent,/, '', true);
+        this.replaceContent(adminModulePath, /\s*EntityAuditService,/, '', true);
+
+        // add to import statement
+        this.replaceContent(
+          adminModulePath,
+          /(UserModalService),?(\n} from)/m,
```

Comment:
this will break if the UserModalService import changes, may be add a needle or add it before the @NgModule directive with a space after

Reply:
Even better would be to contain everything related to this in its own module and just import the module here

Reply:
Quick question regarding this. JHipster appears to have a slightly different convention for dealing with routes in a module than Angular CLI (which creates a separate `*RoutingModule`). Are there plans to change JHipster to match the CLI conventions? Is it better to follow them, or just stick to what's in JHipster at the moment?